### DISCUSSION TITLE
How to forbid delete of all items inside a collection?

### DISCUSSION THREAD

**Discussion by GalinhaLX** [mavoweb/mavo#813] [Q&A] [answered]
Hi

Inside a collection (without "add new item to collection" button), 

how to:

Option 1:      Display a button in case items =0
Option 2:     forbid to delete all items?

I know there's `mv-initial-items="1"` , but it doesn't provide this function of 'null value not allowed' 

I've been trying to do option 1 here: [https://codepen.io/GATIGA/pen/MWERdzP](https://codepen.io/GATIGA/pen/MWERdzP)
but it doesn't work, what am I missing?

Thanks for your help

**Comment by GalinhaLX** [comment]
Based on @LeaVerou 's work about "shopping List" ,(included on my codepen for testing)

button appears when I delete text inside the input (not with delete item-button)

unfortunately, this is not what I am looking for

Any idea on how to achieve this?

**Comment by DmitrySharabin** [comment, marked as answer]
Hey there, 👋🏻

I believe we can approach this in different ways. Here are the ones I came up with: https://codepen.io/dmitrysharabin/pen/yLzWyxg?editors=1100. Did I get you correctly?

Please, note that there I'm using the new (upcoming) syntax for collections. Let me know if you want me to build this pen using the stable version of Mavo.

However, it seems to me, something is wrong with `mv-if`: the expression isn't being recalculated (the `Add to list` button doesn't disappear after an item is added to the list; the `Delete` button doesn't reappear when the list has more than one item).


@LeaVerou could you please take a look at whether it's a bug or I'm doing something wrong?

**Reply by LeaVerou** [reply]
It's not that it's not updating, it's that empty values are ignored when doing `count()`, so it returns 0. If you add some text, the button disappears. If you want to count all values, empty or not, you can use `list.length`.

**Reply by DmitrySharabin** [reply]
Oh. I see. Will keep it in mind. Thanks! 🙏🏻

**Reply by GalinhaLX** [reply]
Thank you so much @DmitrySharabin and @LeaVerou 

> Hey there, 👋🏻
> 
> I believe we can approach this in different ways. Here are the ones I came up with: https://codepen.io/dmitrysharabin/pen/yLzWyxg?editors=1100. Did I get you correctly?


Sometimes simple solutions, are so simple, that I even forget about it : hide with CSS 

Seems good option, but I'm interested into second one to learn how it works too with MAVO


> 
> Please, note that there I'm using the new (upcoming) syntax for collections. Let me know if you want me to build this pen using the stable version of Mavo.

don't waste your time, I'm fine with it.
I'll translate it to stable version (so like that I'll practice and learn) and ...great news to know there's an upcoming syntax!


> 
> However, it seems to me, something is wrong with `mv-if`: the expression isn't being recalculated (the `Add to list` button doesn't disappear after an item is added to the list; the `Delete` button doesn't reappear when the list has more than one item).
> 
> @LeaVerou could you please take a look at whether it's a bug or I'm doing something wrong?

I was not able to make it work with `list.length`  as @LeaVerou mentioned
(but with this comment I learned that I can use javascript inside mavo ...as it's javascript also! of course, wow!
sorry, I just passed from mavo maturity level -10 to -9 ...keep going)

So I keep testing your code, and find out the solution with minor corrections on mv-if="count()"

here's the codepen working as I want:

[https://codepen.io/GATIGA/pen/RwLmZJE](https://codepen.io/GATIGA/pen/RwLmZJE)

Thanks to both of you !
